Realm/RLMSyncPermissionValue.mm

```diff
@@ -76,27 +78,41 @@ - (instancetype)initWithRealmPath:(NSString *)path
         _accessLevel = accessLevel;
         _path = path;
         _userID = userID;
+        if (!_userID) {
```

Comment:
I think the `nonnull` declaration should be good enough to demonstrate to users what this initializer requires. I'll remove the check.

> I'd also be inclined to write this as argument validation done before the init stuff rather than validation of the object's initialized state

Yeah, I agree this is cleaner. I'll do that in the future.